### PR TITLE
Add macOS runner to GitHub Action Workflow

### PR DESCRIPTION
**Pull request by RJ722**
## Summary of the changes in this pull request

@jendrikseipp and me discussed setting up an automated release for mac. I want to take it up, but I am still starting out with app development, and have no idea how to create a `dmg`/`pkg` file for distribution.

However, as a start, we are planning to test all new pull requests, and setting up nightly builds to run on mac as well.

## Pull request checklist

- [ ] I have added an entry in `CHANGELOG.md` including my name and issue and/or pull request number.
- [x] If applicable: I have removed the corresponding entry in `TODO.md`.

### CODE REVIEW

**Comment by RJ722** [comment]
Looks like support for macOS Big Sur is still in preview, and the job not starting is a [known issue](https://github.com/actions/virtual-environments/issues/1814#issuecomment-757479694).

I'll go ahead and remove it for now.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # Unreleased
 * Add a "Give Feedback" button (#551, Rahul Jha).
+* Start testing code on macOS (#552, Rahul Jha).
```
Comment (jendrikseipp) [inline CHANGELOG.md:3]: Start testing -> Test

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -31,11 +31,23 @@ jobs:
...
         sudo apt-get -y install -qq gir1.2-gdkpixbuf-2.0 gir1.2-glib-2.0 gir1.2-gtk-3.0 gir1.2-pango-1.0 gir1.2-webkit2-4.0 python3 python3-enchant python3-gi python3-yaml tox
 
+    - name: Install dependencies (macOS)
+      if: startsWith(matrix.os, 'macos')
+      run: |
+        brew install gtk
+        brew install pygobject3 glib libffi cairo gtksourceview3
+        brew install atk librsvg
+        brew install gsettings-desktop-schemas adwaita-icon-theme
+        brew install enchant
```
Comment (jendrikseipp) [inline .github/workflows/run-tests.yml:47]: Can we combine this into one brew command, i.e., on one line?

**Review by jendrikseipp — CHANGES_REQUESTED** [review]
Nice! Looks good to me. I'm surprised that so few changes are necessary for this.

File: .github/workflows/run-tests.yml (hunk 2 above)
Comment (RJ722) [reply .github/workflows/run-tests.yml:47]: I've pushed another commit to check this. Let's see! 🤞

Comment (RJ722) [reply .github/workflows/run-tests.yml]: It seems to be working! \o/

**Comment by jendrikseipp** [comment]
Thanks, Rahul!

**Pull request merged**